sys-kernel/ugrd/ugrd-1.13.0.ebuild

```
@@ -0,0 +1,49 @@
+# Copyright 2023-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{11..12} )
+inherit distutils-r1 optfeature shell-completion
+
+DESCRIPTION="Python based initramfs generator with TOML defintions"
+HOMEPAGE="https://github.com/desultory/ugrd"
+SRC_URI="https://github.com/desultory/${PN}/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	app-misc/pax-utils
+	>=dev-python/zenlib-2.1.2[${PYTHON_USEDEP}]
+	>=dev-python/pycpio-1.2.1[${PYTHON_USEDEP}]
+	sys-apps/pciutils
+"
+
+src_install() {
+	# Call the distutils-r1_src_install function to install the package
+	distutils-r1_src_install
+	# Create the ugrd config directory
+	keepdir /etc/ugrd
+	# Install the example config into /etc/ugrd/config.toml
+	# Do not overwrite an existing config
+	insinto /etc/ugrd
+	newins examples/example.toml config.toml
+	# Create the kernel preinst.d directory if it doesn't exist
+	# Install the kernel preinst.d hook
+	exeinto /usr/lib/kernel/preinst.d
+	doexe hooks/installkernel/52-ugrd.install
+	exeinto /usr/lib/kernel/install.d
+	doexe hooks/kernel-install/52-ugrd.install
+	dobashcomp completion/ugrd  # Install bash autocomplete script
+	dozshcomp completion/_ugrd  # Install zsh autocomplete script
+}
```

```suggestion
python_install_all() {
	distutils-r1_python_install_all
	
	# Install the example config into /etc/ugrd/config.toml
	insinto /etc/ugrd
	newins examples/example.toml config.toml
	
	# Install the kernel installation hooks
	exeinto /usr/lib/kernel/preinst.d
	doexe hooks/installkernel/52-ugrd.install
	exeinto /usr/lib/kernel/install.d
	doexe hooks/kernel-install/52-ugrd.install
	
	dobashcomp completion/ugrd  # Install bash autocomplete script
	dozshcomp completion/_ugrd  # Install zsh autocomplete script
}
```

We don't have to `keepdir` since we `insinto` that dir, `keepdir` is for installing empty directories.

And we should define this as `python_install_all` instead of modifying the eclass's `src_compile`.